File: Ser2023/Address/Address.Web/Modules/Address/Country/CountryRow.cs
using Address.Administration.Entities;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Address.Address
{
    [ConnectionKey("Address"), Module("Address"), TableName("[address].[Country]")]
    [DisplayName("Country"), InstanceName("Country")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]
    public sealed class CountryRow : LoggingRow<CountryRow.RowFields>, IIdRow, INameRow
    {
        [DisplayName("Country Code"), Size(3), PrimaryKey, NotNull, IdProperty, QuickSearch, NameProperty]
        public string CountryCode
        {
            get => fields.CountryCode[this];
            set => fields.CountryCode[this] = value;
        }

        [DisplayName("Name"), Size(50), NotNull]
        public string Name
        {
            get => fields.Name[this];
            set => fields.Name[this] = value;
        }

        [DisplayName("Is Active"), NotNull]
        public short? IsActive
        {
            get => fields.IsActive[this];
            set => fields.IsActive[this] = value;
        }

        public CountryRow()
            : base()
        {
        }

        public CountryRow(RowFields fields)
            : base(fields)
        {
        }

        public class RowFields : LoggingRowFields
        {
            public StringField CountryCode;
            public StringField Name;
            public Int16Field IsActive;
        }
    }
}

File: Ser2025/Company/Company.Web/Modules/ServerTypes/Company/CountryRow.ts
import { fieldsProxy } from '@serenity-is/corelib';

export interface CountryRow {
    CountryCode?: string;
    CountryName?: string;
    IsActive?: number;
    TenantId?: number;
}

export abstract class CountryRow {
    static readonly idProperty = 'CountryCode';
    static readonly nameProperty = 'CountryCode';
    static readonly localTextPrefix = 'Company.Country';

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<CountryRow>();
}